iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -81,69 +134,51 @@ class Sample
     operator bool() const;
 
     ///
-    /// @brief allocation Access to the memory chunk loaned to the sample.
-    /// @return
+    /// @brief allocation Access to the encapsulated type loaned to the sample.
+    /// @return a pointer to the encapsulated type.
+    /// @details Only available for non-const type T.
     ///
+    template <typename S = T, typename = std::enable_if_t<std::is_same<S, T>::value && !std::is_const<S>::value, S>>
```

Comment:
Is `static_assert` possible?

Reply:
in this case the compiler would probably choose the non-const method in a non-const environment, which triggers the static_assert and makes it awkward to use the correct method